### DISCUSSION TITLE
Source .zshrc?

### DISCUSSION THREAD

**Discussion by dpo** [wezterm/wezterm#3255] [General]
Hi,

How do you get wezterm to source `~/.zshrc` at startup and when opening a new tab?

Thanks!

**Comment by nekowinston** [comment]
That's your shell's job, not WezTerms. Just make sure that `zsh` is your default shell. Have a look at the [`zsh` manpage](https://linux.die.net/man/1/zsh) "STARTUP/SHUTDOWN FILES" section.

**Reply by dpo** [reply]
Thanks. For some reason, `$HOME/.zshrc` isn't sourced automatically on macOS (at least for me). It seems to be ok now that I created `$HOME/.zprofile`, which contains `source $HOME/.zshrc`.

**Reply by nekowinston** [reply]
That's pretty odd, I'm on macOS as well - `~/.zshenv` is the only file in my home directory.
You normally shouldn't source your `.zshrc` (config files for interactive shell sessions) in your `.zshenv`/`.zprofile`. IIRC you'd only have to explicitly source `zshrc` if your session isn't interactive.
What's your output in a new WezTerm session when you run `wezterm.mux.get_pane(0):get_foreground_process_info()` in the debug overlay? (Ctrl+Shift+L)

**Reply by wez** [reply]
Note that wezterm runs shells as login shells by default, which affects which startup files the shell loads.

**Reply by dpo** [reply]
This has been a bit of a mystery since I've started using zsh on macOS. A quick Google search reveals it's a common gripe. In iTerm2, I also had to explicitly add `source ~/.zshrc` as the first command the terminal sends to the shell.

Here's the output you asked for:
```
Debug Overlay
wezterm version: 20221119-145034-49b9839f
OpenGL version: Apple M1 Max 4.1 Metal - 76.3
Enter lua statements or expressions and hit Enter.
Press ESC or CTRL-D to exit
> wezterm.mux.get_pane(0):get_foreground_process_info()
{
    "argv": [
        "-zsh",
    ],
    "children": {},
    "cwd": "/Users/dpo",
    "executable": "/bin/zsh",
    "name": "zsh",
    "pid": 95382,
    "ppid": 95381,
    "start_time": 1679166845,
    "status": "Run",
}
```

**Reply by dpo** [reply]
I was able to make sense of things thanks to https://zsh.sourceforge.io/Guide/zshguide02.html#l9
I moved all `PATH` settings to `.zshenv` and left aliases, function definitions, prompt settings, and options in `.zshrc`, and now things appear to be working correctly. Sorry for this discussion that's unrelated to wezterm. It helped me begin to understand zsh though.

**Comment by gchingon** [comment]
I'm a prose writer on a Mac dumb enough to prefer to write in neovim, to better understand what I'm doing "under the hood" I've broken up my functions into separate files for copy.zsh (copy Y filetype to Z dir), move.zsh (move X filetype to specific dir), nvim.zsh (open configs for nvim, wez, etc) 

so it's .config/zsh/functions/copy.zsh, move.zsh, nvim.zsh, etc.

but I always seem to have 'source .zshrc' in every new window/tab.

I've tried adding `source $HOME/.zshrc` and `source /Users/username/.zshrc` in `.zprofile` and still I have to type out `source .zshrc` on every new window/tab

has anyone on MacOS figured this out? if so can it be added to doc files?
